timemachine/md/minimizer.py

```diff
@@ -395,22 +389,23 @@ def equilibrate_host(
     group_indices = get_group_indices(bond_list, len(combined_masses))
 
     barostat_interval = 5
-    u_impls = bind_potentials(parameterize_system(hgt, ff, 0.0))  # lambda=0
+    _, params = parameterize_system(hgt, ff, 0.0)  # lambda=0
+    u_impl.set_params(flatten_params(params))
     barostat = MonteCarloBarostat(
         x0.shape[0],
         pressure,
         temperature,
         group_indices,
         barostat_interval,
         seed,
-    ).impl(u_impls)
+    ).impl([u_impl])
 
     # context components: positions, velocities, box, integrator, energy fxns
-    ctxt = custom_ops.Context(x0, v0, host_config.box, integrator, u_impls, barostat)
+    ctxt = custom_ops.Context(x0, v0, host_config.box, integrator, [u_impl], barostat)
 
-    ctxt.multiple_steps(n_steps)
+    xs, boxes = ctxt.multiple_steps(n_steps)
 
-    return ctxt.get_x_t(), ctxt.get_box()
+    return xs[-1], boxes[-1]
 
 
 def get_val_and_grad_fn(bps: Iterable[BoundPotential], box: NDArray, precision=np.float32):
```

Comment:
possibly later: consider consolidating `summed_potential_from_potentials_and_params`, `make_host_du_dx_fxn`, and `get_val_and_grad_fn` in this file.

(Maybe body of `get_val_and_grad_fn` can be slightly simplified using `summed_potential_from_potentials_and_params`.)